server/app/views/applicant/NorthStarApplicantProgramBlockEditView.java

```diff
@@ -112,10 +113,36 @@ public String render(Request request, ApplicationBaseViewParams applicationParam
       context.setVariable(
```

Comment:
**suggestion:** This is a bit hard to parse. I think one way to make this a little more clear would be to create two functions `setContextForReview` or `setContextForPrevious` and then you can branch which function gets called based on the applicationParams.errorDisplayMode(). And beyond that you could even make a helper function `setContextForFormModal`, which gets called from each of those sub-functions and would take in variables, like errorModalFormAction, errorModalTitle, errorModalContent, errorModalButtonText and errorModalDataRedirectTo and set the context for each. And then each function would set their particular form action then call the helper function (if you think its cleaner to create that)

Reply:
thanks for the suggestion.  refactored the code. PTAL